### PR TITLE
Refactor code to avoid duplicate retry on downloading segment to local

### PR DESCRIPTION
**Pull request by jackjlli**
This PR removes the outer retry wrapper code. That's because:

1. `downloadSegmentToLocal` method has already used retry policy to fetch segment to local M times (M = 3 by default). If we keep the outer retry wrapper, which retries at most N times(N = 5 by default), the total retry times would be M*N.

2. if an exception is thrown when untarring(right after the method `downloadSegmentToLocal` is called), that means the tar file is broken or not found AFTER the retry. There's no need to do it again. Simply throw the exception would be good.

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ public ServerType getServerType() {
     public static final long DEFAULT_QUERY_EXECUTOR_TIMEOUT_MS = 15_000L;
     public static final String DEFAULT_REQUEST_HANDLER_FACTORY_CLASS =
         "com.linkedin.pinot.server.request.SimpleRequestHandlerFactory";
-    public static final String DEFAULT_SEGMENT_LOAD_MAX_RETRY_COUNT = "5";
+    public static final String DEFAULT_SEGMENT_LOAD_MAX_RETRY_COUNT = String.valueOf(SegmentFetcher.RETRY_DEFAULT);
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java]: Don't use the value from SegmentFetcher, it's separate config. Recommend change the type to int

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java [hunk 2]
```diff
@@ -61,7 +62,9 @@ public SegmentFetcherAndLoader(DataManager dataManager, ZkHelixPropertyStore<ZNR
     } catch (Exception e) {
       // Keep the default value
     }
-    _segmentLoadMaxRetryCount = maxRetries;
+    if (maxRetries != Integer.parseInt(CommonConstants.Server.DEFAULT_SEGMENT_LOAD_MAX_RETRY_COUNT)) {
```
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: int maxRetries = pinotHelixProperties.getInt(CommonConstants.Server.CONFIG_OF_SEGMENT_LOAD_MAX_RETRY_COUNT, CommonConstants.Server.DEFAULT_SEGMENT_LOAD_MAX_RETRY_COUNT);

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java [hunk 3]
```diff
@@ -71,15 +74,18 @@ public SegmentFetcherAndLoader(DataManager dataManager, ZkHelixPropertyStore<ZNR
     } catch (Exception e) {
       // Keep the default value
     }
-    _segmentLoadMinRetryDelayMs = minRetryDelayMillis;
+    if (minRetryDelayMillis != Integer.parseInt(CommonConstants.Server.DEFAULT_SEGMENT_LOAD_MIN_RETRY_DELAY_MILLIS)) {
```
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: Same here after changing the data type of CommonConstants.Server.DEFAULT_SEGMENT_LOAD_MIN_RETRY_DELAY_MILLIS to long

**Comment by jackjlli** [comment]
@Jackie-Jiang RetryPolicy has already been used in `fetchSegmentToLocal` method. Thus, there's no need to reuse it in this part.

**Pull request merged**